pkg/render/monitor/monitor.go

```diff
@@ -1199,3 +1232,181 @@ func (mc *monitorComponent) serviceMonitorCalicoKubeControllers() *monitoringv1.
 		},
 	}
 }
+
+// externalPrometheusRole creates the permissions for the external prometheus server to scrape ours.
+func (mc *monitorComponent) externalPrometheusRole() client.Object {
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TigeraExternalPrometheus,
+			Namespace: mc.cfg.Monitor.ExternalPrometheus.Namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				// When prometheus was first introduced it was accessed through k8s services/proxy and so to this day,
+				// the following resources are used to authorize access to the prometheus metrics.
+				APIGroups: []string{""},
+				Resources: []string{"services/proxy"},
+				ResourceNames: []string{
+					"https:tigera-api:8080", "calico-node-prometheus:9090",
+				},
+				Verbs: []string{"get", "create"},
+			},
+		},
+	}
+}
+
+// externalPrometheusRoleBinding creates the permissions for the external prometheus server to scrape ours.
+func (mc *monitorComponent) externalPrometheusRoleBinding() client.Object {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TigeraExternalPrometheus,
+			Namespace: mc.cfg.Monitor.ExternalPrometheus.Namespace,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      TigeraExternalPrometheus,
+				Namespace: mc.cfg.Monitor.ExternalPrometheus.Namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     TigeraExternalPrometheus,
+		},
+	}
+}
+
+// externalPrometheusTokenSecret creates the bearer token on which behalf requests will be made from the external prometheus
+// server to ours.
+func (mc *monitorComponent) externalPrometheusTokenSecret() client.Object {
```

Comment:
https://kubernetes.io/docs/concepts/security/service-accounts/#get-a-token

I think we should be using token projection instead of an explicitly created secret here - is that feasible?

Reply:
If not, we'll probably want to do a follow-on to use the token request API instead of a long-lived token with no expiry.

Reply:
I only see authn/authz options that involve reading a file or a secret in the ServiceMonitorSpec: BearerTokenFile, BearerTokenSecret, Authorization, which relies on a secretKeySelector. So I think today there is not yet an option to use the token api.
We do allow the user to override our BearerTokenSecret, so we would not render this secret, but that is all that I was able to figure out.